html/cross-origin-opener-policy/resources/coop-coep.py

```diff
@@ -21,13 +21,41 @@ def main(request, response):
 <script src="/common/get-host-info.sub.js"></script>
 <iframe></iframe>
 <script>
-  const navigate = new URL(location).searchParams.get("navigate");
-  if (navigate !== null) {
-    self.location = navigate;
+  const params = new URL(location).searchParams;
+  const navHistory = params.get("navHistory");
+  const avoidBackAndForth = params.get("avoidBackAndForth");
+  const navigate = params.get("navigate");
+  const openerDOMAccess = params.get("openerDOMAccess");
+  // Need to wait until the page is fully loaded before navigating
+  // so that it creates a history entry properly.
+  async function fullyLoaded() {
+    return new Promise((resolve, reject) => {
```

Comment:
Maybe `addEventListener('error', reject);`

Reply:
`error` is happens for uncaught exceptions and syntax errors from parsing scripts. I don't think it's useful to reject the promise if that happens. `load` will happen anyway.